### PR TITLE
Fix RESTEasy classic server templating for observability

### LINKED ISSUE #18251: Opentelemetry server templating does not work with template in class Path annotation
## Describe the bug
I'm not sure whether this should even be allowed, but when adding a Path annotation with a template parameter to a class the server span contains the actual path of the request instead of the templated version.

According to @kenfinnigan it is worth figuring out what is supposed to happen here.

### Expected behavior
Either Quarkus should complain about the template parameter in that place or the template should be correctly replaced.

### Actual behavior
The actual url without replacement is used as the name of the server span.

## To Reproduce
The problem is reproducible with a resource similar to:
```java
@Path("/tracing/filter/{dummy}")
public static class TestResource {
  @GET
  public String get(@PathParam("dummy") String dummy) {
    // ...
  }
}
```

The resulting name of the span has the full path of e.g. `/tracing/filter/1234` instead of the template `/tracing/filter/{dummy}`.

in contrast templating is working fine with
```java
@Path("/tracing")
public static class TestResource {
  @GET
  @Path("/filter/{dummy}")
  public String get(@PathParam("dummy") String dummy) {
    // ...
  }
}
```
### Quarkus version or git rev
2.0.0.Final

### PR DESCRIPTION
**Pull request by kenfinnigan**
- Fixes #18251
- Add test to Micrometer integration test. Verified not a problem with RESTEasy Reactive
- Add test to OpenTelemetry integration test to very fix for RESTEasy Classic

### CODE REVIEW

**Comment by kenfinnigan** [comment]
I haven't been able to replicate this failure locally, still trying

**Comment by kenfinnigan** [comment]
Ok, should be fixed now

**Review by ebullient — APPROVED** [review]

File: extensions/resteasy-classic/resteasy/deployment/src/main/java/io/quarkus/resteasy/deployment/RestPathAnnotationProcessor.java [hunk 1]
```diff
@@ -79,21 +81,29 @@ public void transform(TransformationContext ctx) {
...
                 ClassInfo classInfo = methodInfo.declaringClass();
 
-                AnnotationInstance annotation = methodInfo.annotation(REST_PATH);
-                if (annotation == null) {
-                    return;
+                AnnotationInstance classAnnotation = classInfo.classAnnotation(REST_PATH);
+                AnnotationInstance methodAnnotation = methodInfo.annotation(REST_PATH);
+                if (methodAnnotation == null) {
+                    // Check for @Path on class and not method
+                    if (!(classAnnotation != null && isRestEndpointMethod(methodInfo.annotations()))) {
```
Comment (ebullient) [inline extensions/resteasy-classic/resteasy/deployment/src/main/java/io/quarkus/resteasy/deployment/RestPathAnnotationProcessor.java:88]: Is there any way to make this more readable? Lots of nots.. (that's a wish though.. )

Comment (kenfinnigan) [reply extensions/resteasy-classic/resteasy/deployment/src/main/java/io/quarkus/resteasy/deployment/RestPathAnnotationProcessor.java:88]: Hmm, I wonder if dropping the class annotation check completely will work, as that should still hit the case of method without `@Path` but with `@GET`. Will give it a try

Comment (kenfinnigan) [reply extensions/resteasy-classic/resteasy/deployment/src/main/java/io/quarkus/resteasy/deployment/RestPathAnnotationProcessor.java]: Ok, it works. Pushed the change

**Pull request merged**